### PR TITLE
Release v6.11.0 lacework-agent Helm Charts

### PR DESCRIPTION
**Pull request by brooks-connor-a**

### CODE REVIEW

File: lacework-agent/templates/configmap.yaml [hunk 1]
```diff
@@ -37,13 +37,17 @@ data:
...
         },
-       {{- else if or (eq (toString .enable) "experimental") (eq (toString .enable) "false") }}
+       {{- else if (eq (quote .enable) "experimental") }}
         "codeaware": {
-          "enable": {{ quote .enable }}
+          "enable": "experimental"
+        },
+       {{- else if (eq (quote .enable) "false") }}
+        "codeaware": {
+          "enable": "false"
```
Comment (nschmeller) [inline lacework-agent/templates/configmap.yaml:50]: The Helm charts repo is ahead for this, there should be no `codeaware` changes

File: lacework-agent/values.yaml [hunk 2]
```diff
@@ -72,7 +72,7 @@ laceworkConfig:
   # [Optional] Enable Active Vulnerability Detection
   # https://docs.lacework.net/onboarding/configure-agent-behavior-in-configjson-file#codeaware-property
   codeaware:
-    enable:
+    enable: false
```
Comment (nschmeller) [inline lacework-agent/values.yaml:75]: This hunk should also not be included for the same reason

**Review by nschmeller — CHANGES_REQUESTED** [review]
Some of these changes have already been included in the Helm charts, so merging this would be a regression.

**Review by nschmeller — APPROVED** [review]
Thanks for handling this!

File: lacework-agent/templates/configmap.yaml [hunk 3]
```diff
@@ -54,14 +54,15 @@ data:
...
+        "ebpf": {
+          "shortlivedmsecs": "{{ .shortlivedmsecs }}"
+        },
+       {{- end }}
      {{- end }}
-     },
      {{- with .Values.laceworkConfig.fim }}
-     {{- if .enable }}
+     {{- if (kindIs "bool" .enable) }}
+       {{- if .enable }}
```
Comment (joebowbeer) [inline lacework-agent/templates/configmap.yaml:65]: @brooks-connor-a With this change merged, `fim:` is once again required.

and in the following values snippet, I needed to add an explicit `enable: true` or else fim would be disabled.

```yaml
fim:
  enable: true
```

Related #72 #96

Comment (nschmeller) [reply lacework-agent/templates/configmap.yaml:65]: Hey @joebowbeer, FIM is still enabled by default with this change as it is the agent's default behavior. This is the case even if you don't set `fim.enable` in your `values.yaml`. 

To give a little more context, before this change, FIM was always explicitly enabled for every agent deployed with Helm, which had the unintended consequence of [blocking config changes made from the Lacework console](https://docs.lacework.net/onboarding/configure-agent-behavior-in-console#how-configuring-the-agent-from-the-lacework-console-works). After this change, users still have the ability to explicitly enable / disable FIM via the Helm values file, but if users do not set in the values file, the agent will still enable FIM unless it is turned off via server-side config. 

Hope that helps!

Comment (joebowbeer) [reply lacework-agent/templates/configmap.yaml:65]: Thanks for the explanation. If the agent's default fim mode is enable, then I don't see a problem.

I provide more details below for completeness, anyway.

I'm invoking `helm template` with lacework as a subchart:

Chart.yaml

```yaml
apiVersion: v2
name: lacework
version: "0"
dependencies:
- name: lacework-agent
  version: "6.12.0"
  repository: https://lacework.github.io/helm-charts/
annotations:
  release: lacework
  namespace: lacework
```

If I exec `helm template .` with the following values.yaml:

```yaml
lacework-agent:
  laceworkConfig:
    accessToken: "redacted"
```

I get this error message:

```
Error: values don't meet the specifications of the schema(s) in the following chart(s):
lacework-agent:
- laceworkConfig.fim: enable is required
```

If I exec `helm template .` with the following values.yaml:

```yaml
lacework-agent:
  laceworkConfig:
    accessToken: "redacted"
    fim:
```

I get this warning:

```
coalesce.go:286: warning: cannot overwrite table with non table for lacework.lacework-agent.laceworkConfig.fim (map[coolingPeriod:<nil> crawlInterval:<nil> enable:<nil> fileIgnore:[] filePath:[] noAtime:<nil> runAt:<nil>])
```

and there is no `fim.mode: enable` in the rendered config.json

I needed to use the following values.yaml to restore `fim.mode: enable` to config.json

```yaml
lacework-agent:
  laceworkConfig:
    accessToken: "redacted"
    fim:
      enable: true
```

Comment (nschmeller) [reply lacework-agent/templates/configmap.yaml:65]: Great callout @joebowbeer, I opened https://github.com/lacework/helm-charts/pull/235 to fix the requirement for `fim.enable` by the values schema. Thanks!

**Pull request merged**